go/database/mpt/proof_test.go

```diff
@@ -0,0 +1,462 @@
+			t.Errorf("unexpected hash: got %x, want %x", got, want)
+		}
+	}
+
+	if got, want := len(proof), len(nodes); got != want {
+		t.Fatalf("unexpected proof length: got %d, want %d", got, want)
+	}
+}
+
+func TestCreateAccountProof_CannotGenerateProof(t *testing.T) {
+	ctrl := gomock.NewController(t)
+
+	nodes := []Node{&EmptyNode{}}
+
+	nodeSource := mockNodeSource(ctrl, nodes)
+	root := NewNodeReference(EmptyId())
+	source := NewProofSource(&root, nodeSource)
+
+	proof, err := source.CreateAccountProof(common.Address{0xA})
+	if err != nil {
+		t.Fatalf("failed to create account proof: %v", err)
+	}
+
+	if got, want := len(proof), 0; got != want {
+		t.Errorf("unexpected proof length: got %d, want %d", got, want)
+	}
+}
+
+func TestCreateStorageProofs_CanGenerateManyProofs(t *testing.T) {
+	ctrl := gomock.NewController(t)
+
+	nodes := []Node{&BranchNode{}, &ExtensionNode{}, &AccountNode{address: common.Address{0xA}}}
+
+	key1 := common.Key{0x1}
+	key2 := common.Key{0x2}
+
+	storageNodes1 := []Node{&BranchNode{}, &ExtensionNode{}, &ValueNode{key: key1}}
+	storageNodes2 := []Node{&ExtensionNode{}, &BranchNode{}, &BranchNode{}, &ValueNode{key: key2}}
+	storageNodes := [][]Node{storageNodes1, storageNodes2}
+
+	nodeSource := mockNodeSource(ctrl, append(nodes, append(storageNodes1, storageNodes2...)...))
+	hasher := nodeSource.getConfig().Hashing.createHasher()
+	root := NewNodeReference(EmptyId())
+	source := NewProofSource(&root, nodeSource)
+
+	proof, err := source.CreateStorageProofs(common.Address{0xA}, key1, key2)
+	if err != nil {
+		t.Fatalf("failed to create account proof: %v", err)
+	}
+
+	// check account proof is present
+	for i, element := range proof.GetAccountProof() {
+		rlp, err := hasher.encode(nodes[i], nodeSource, []byte{})
+		if err != nil {
+			t.Fatalf("failed to encode node: %v", err)
+		}
+		if got, want := element.Rlp(), rlp; !bytes.Equal(got, want) {
+			t.Errorf("unexpected RLP: got %x, want %x", got, want)
+		}
+		if got, want := element.Hash(), common.Keccak256(rlp); got != want {
+			t.Errorf("unexpected hash: got %x, want %x", got, want)
+		}
+	}
+
+	// continue checking storage proofs are present
+	for i, key := range []common.Key{key1, key2} {
+		for j, element := range proof.GetStorageProof(key) {
+			rlp, err := hasher.encode(storageNodes[i][j], nodeSource, []byte{})
+			if err != nil {
+				t.Fatalf("failed to encode node: %v", err)
+			}
+			if got, want := element.Rlp(), rlp; !bytes.Equal(got, want) {
+				t.Errorf("unexpected RLP: got %x, want %x", got, want)
+			}
+			if got, want := element.Hash(), common.Keccak256(rlp); got != want {
+				t.Errorf("unexpected hash: got %x, want %x", got, want)
+			}
+		}
+
+		if got, want := len(proof.GetStorageProof(key)), len(storageNodes[i]); got != want {
+			t.Errorf("unexpected proof length: got %d, want %d", got, want)
+		}
+	}
+}
+
+func TestCreateStorageProofs_CannotGenerateProofs(t *testing.T) {
```

Comment:
"CannotGenerateProofs" ... in what situation should it not be able to generate proofs?

Ideally test names have the format "Test<Subject>_<Condition><ExpectedOutcome>". Here, for instance, it could be something like
```
TestCreateStorageProofs_NonExistingStorageSlotsLeadToEmptyProofs"
```